### PR TITLE
[docs] Fix layout jump issue

### PR DESCRIPTION
**Pull request by oliviertassinari**
Also rename `Sticky columns` into `Column pinning`.

### CODE REVIEW

File: docs/src/pages/components/data-grid/rows/ServerSortingGrid.js [hunk 1]
```diff
@@ -69,20 +69,18 @@ export default function ServerSortingGrid() {
...
     <div style={{ height: 400, width: '100%' }}>
-      <DataGrid
-        rows={rows}
-        columns={data.columns}
-        sortingMode="server"
-        sortModel={sortModel}
-        onSortModelChange={handleSortModelChange}
-        loading={loading}
-      />
+      {data.columns.length === 0 ? null : (
```
Comment (dtassone) [inline docs/src/pages/components/data-grid/rows/ServerSortingGrid.js:74]: there should be the no rows layout

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/rows/ServerSortingGrid.js:74]: It crashes

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/rows/ServerSortingGrid.js:74]: It's likely a bug, I can open a new issue to keep track of it.

**Comment by oliviertassinari** [comment]
The layout jump is problematic when using anchor links, the scroll end-up in the wrong position.

File: docs/src/pages/components/data-grid/rows/ServerSortingGrid.js (hunk 1 above)
Comment (dtassone) [reply docs/src/pages/components/data-grid/rows/ServerSortingGrid.js:74]: FYI We have a no-rows-no-cols story in storybook `datasets`

Comment (dtassone) [reply docs/src/pages/components/data-grid/rows/ServerSortingGrid.js:74]: https://deploy-preview-338--material-ui-x.netlify.app/storybook/?path=/story/x-grid-tests-dataset--no-rows-no-cols

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/rows/ServerSortingGrid.js:74]: Seems the problem is gone. No idea. I'm removing the defensive logic.

**Review by dtassone — APPROVED** [review]

**Pull request merged**